### ISSUE TITLE
Bug: test_metrics is overwritten by val_metrics during evaluation

### ISSUE THREAD

**Issue by yingziyu-llt** [GSanShui/XPert#6]
Hi XPert authors, thanks for the great work!

While running the model and evaluating the 5-fold cross-validation results, I noticed that the metrics for the test set are completely identical to the validation set in the final output CSV.

Upon inspecting the source code, I found a hardcoded assignment in `train_xpert.py` that overwrites the true test metrics.

**Bug Location:**

In `train_xpert.py`, inside the `main()` function under the `train` mode branch (around line 344):

```python
logger.info('Tesing on test dataset:')
_,_,_,_,_, test_metrics = validate(model, test_dataloader, args, config, flag=True, scaler=gradscaler)

test_metrics = val_metrics  # <--- Issue is here

logger.info('Test Metrics:')
log_nested_dict(test_metrics, logger, indent=1)  

```

Because of `test_metrics = val_metrics`, the true test results are discarded, and the CSV logs the validation metrics twice.

Additionally, a similar redundant assignment exists under the `test` mode branch (around line 382):

```python
test_metrics = test_metrics
val_metrics = test_metrics
tr_metrics = test_metrics

```

**Proposed Fix:**

Removing the `test_metrics = val_metrics` line in the `train` mode logic will resolve this issue and allow the model to output the correct test set performance at the end of training.
